app/client/src/pages/Editor/DataSourceEditor/NewActionButton.tsx

```diff
@@ -36,6 +36,22 @@ interface NewActionButtonProps {
   style?: any;
   isNewQuerySecondaryButton?: boolean;
 }
+
+export const isApiPluginInvalidUrl = (
+  currentEnvironment: string,
+  pluginType?: string,
+  datasource?: Datasource,
+) => {
+  return (
+    pluginType === PluginType.API &&
+    (!datasource ||
+      !datasource.datasourceStorages[currentEnvironment]
+        ?.datasourceConfiguration ||
+      !datasource.datasourceStorages[currentEnvironment]
+        ?.datasourceConfiguration?.url)
+  );
+};
```

Comment:
Code we make this code more readable?

```suggestion
   if(pluginType !== PluginType.API) return false; 
   if(!datasource) return false 
   return !datasource?.datasourceStorages[currentEnvironment]
        ?.datasourceConfiguration?.url
  
};
```